samples/bluetooth/peripheral_hids_mouse/src/main.c

```diff
@@ -140,13 +215,25 @@ static void connected(struct bt_conn *conn, u8_t err)
 		return;
 	}
 
+	bool is_conn_slot_free = false;
+	bool is_conn_assigned = false;
+
 	for (size_t i = 0; i < CONFIG_BT_GATT_HIDS_MAX_CLIENT_COUNT; i++) {
-		if (!conn_mode[i].conn) {
```

Comment:
Maybe split this into two?
Inserting the conn object, and checking if a free object exists.
we could create two simple helper functions for this.

Reply:
yes, good idea, code would be more clear this way